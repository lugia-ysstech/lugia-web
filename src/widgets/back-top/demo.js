/**
 *
 * create by guorg
 *
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';
import BackTop from './back-top';
import Widget from '../consts/index';
import Theme from '../theme';
import { getBorder } from '@lugia/theme-utils';
import { getBorderRadius } from '../theme/CSSProvider';

const Demo = styled.div`
  & > div > div {
    bottom: ${props => 50 + props.cur * 50}px;
  }
`;
const DemoBack = styled.div`
  width: 40px;
  height: 40px;
  line-height: 40px;
  text-align: center;
  border-radius: 4px;
  border: 1px solid #e8e8e8;
`;
export default class AffixDemo extends React.Component<any, any> {
  EleRef: any;
  render() {
    const view = {
      [Widget.BackTop]: {
        Container: {
          normal: {
            color: 'red',
            background: { color: 'orange' },
            width: 100,
            height: 100,
            opacity: 0.7,
            border: getBorder({ color: 'red', width: 1, style: 'solid' }),
            borderRadius: getBorderRadius(100),
          },
        },
        BackTopIcon: {
          normal: {
            color: 'green',
            fontSize: 16,
            padding: {
              right: 10,
              left: 10,
              top: 10,
              bottom: 10,
            },
            margin: {
              right: 0,
              left: 0,
              top: 0,
              bottom: 0,
            },
          },
        },
      },
    };

    return (
      <div style={{ height: '1600px' }}>
        <Demo cur={1}>
          <Theme>
            <BackTop showType="textType" text="up" />
          </Theme>
        </Demo>
        <Demo cur={2}>
          <Theme config={view}>
            <BackTop icon={'lugia-icon-logo_apple'} showType="iconType" />
          </Theme>
        </Demo>
        <Demo>
          <Theme>
            <BackTop />
          </Theme>
        </Demo>
        <Demo cur={4}>
          <BackTop>
            <DemoBack>UP</DemoBack>
          </BackTop>
        </Demo>
        <div
          style={{ width: '200px', height: '200px', overflowY: 'scroll', marginTop: '400px' }}
          ref={node => (this.EleRef = node)}
        >
          <Demo cur={4}>
            <div style={{ width: '20px', height: '300px' }} />
            <BackTop
              showType="textType"
              text="up"
              visibilityHeight={80}
              target={() => this.EleRef}
            />
          </Demo>
        </div>
        <br />
        <br />
        <h2>injectLugiad属性, 展示使用</h2>
        <div style={{ margin: '20px' }}>
          <BackTop showType="textType" text="up" injectLugiad={{ type: 'BackTop' }} />
        </div>
      </div>
    );
  }
}
