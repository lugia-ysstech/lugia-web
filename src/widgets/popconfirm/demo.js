/**
 *
 * create by liangguodong on 2018/12/5
 *
 * @flow
 */
import * as React from 'react';
import Direction from '../button';
import Icon from '../icon';
import Switch from '../switch';
import notification from '../notification';
import styled from 'styled-components';
import Popconfirm from './popconfirm';
import Input from '../input/index';
import Widget from '../consts';
import Theme from '../theme';

const Wrapper = styled.div`
  margin: 100px;
`;
export class Condition extends React.Component<any, any> {
  state = {
    visible: false,
    condition: true,
  };

  changeCondition = (value: Object) => {
    const condition = value.oldValue === true;
    this.setState({ condition });
  };

  confirm = () => {
    this.setState({ visible: false });
    notification.success({ title: '操作成功 ' });
  };

  cancel = () => {
    this.setState({ visible: false });
    notification.error({ title: '取消操作' });
  };

  handleVisibleChange = (visible: Object) => {
    if (!visible) {
      this.setState({ visible });
      return;
    }
    if (this.state.condition === true) {
      this.confirm();
    } else {
      this.setState({ visible });
    }
  };

  render() {
    return (
      <div style={{ margin: 20 }}>
        <p> 触发弹出框</p>
        <Switch defaultChecked onChange={this.changeCondition} />
        <Popconfirm
          title="确定要删除吗?"
          visible={this.state.visible}
          onVisibleChange={this.handleVisibleChange}
          onConfirm={this.confirm}
          onCancel={this.cancel}
          okText="确定"
          cancelText="取消"
        >
          <Direction>删除任务</Direction>
        </Popconfirm>
      </div>
    );
  }
}

export const WrapperDemo = () => {
  const text = '确定删除这个选项吗?';
  const config = {
    [Widget.Button]: { Container: { normal: { width: 80 } } },
  };
  const iconConfig = {
    [Widget.Icon]: {
      normal: {
        fontSize: 14,
        color: 'red',
      },
    },
  };
  return (
    <Wrapper>
      <Theme config={config}>
        <div style={{ marginLeft: 80, whiteSpace: 'nowrap' }}>
          <Popconfirm placement="topLeft" title={text} action={'click'}>
            <Direction type="primary">TL</Direction>
          </Popconfirm>
          <Popconfirm placement="top" title={text}>
            <Direction type="primary">Top</Direction>
          </Popconfirm>
          <Popconfirm placement="topRight" title={text}>
            <Direction type="primary">TR</Direction>
          </Popconfirm>
        </div>
        <div style={{ width: 70, float: 'left' }}>
          <Popconfirm placement="leftTop" title={text}>
            <Direction type="primary">LT</Direction>
          </Popconfirm>
          <Popconfirm placement="left" title={text}>
            <Direction type="primary">Left</Direction>
          </Popconfirm>
          <Popconfirm placement="leftBottom" title={text}>
            <Direction type="primary">LB</Direction>
          </Popconfirm>
        </div>
        <div style={{ width: 70, marginLeft: 320 }}>
          <Popconfirm placement="rightTop" title={text}>
            <Direction type="primary">RT</Direction>
          </Popconfirm>
          <Popconfirm placement="right" title={text}>
            <Direction type="primary">Right</Direction>
          </Popconfirm>
          <Popconfirm placement="rightBottom" title={text}>
            <Direction type="primary">RB</Direction>
          </Popconfirm>
        </div>
        <div style={{ marginLeft: 80, clear: 'both', whiteSpace: 'nowrap' }}>
          <Popconfirm placement="bottomLeft" title={text}>
            <Direction type="primary">BL</Direction>
          </Popconfirm>
          <Popconfirm placement="bottom" title={text}>
            <Direction type="primary">Bottom</Direction>
          </Popconfirm>
          <Popconfirm placement="bottomRight" title={text}>
            <Direction type="primary">BR</Direction>
          </Popconfirm>
        </div>

        <br />
        <Popconfirm title={text} action={'focus'} placement="bottom">
          <Input placeholder={'聚焦弹出'} />
        </Popconfirm>
        <Popconfirm title={text} action={'hover'} placement="bottom">
          <Direction type="primary"> 悬停</Direction>
        </Popconfirm>
        <Popconfirm
          title={text}
          action={'click'}
          cancelText="No"
          okText="yes"
          okType="danger"
          placement="bottom"
        >
          <Direction type="primary">点击</Direction>
        </Popconfirm>
        <br />
        <div>
          <Popconfirm
            title={text}
            action={'click'}
            cancelText="No"
            okText="yes"
            okType="danger"
            icon={
              <Theme config={iconConfig}>
                <Icon iconClass={'lugia-icon-reminder_exclamation_circle_o'} singleTheme />
              </Theme>
            }
          >
            <Direction type="primary">提示</Direction>
          </Popconfirm>
          <Popconfirm title={text} action={'click'} cancelText="No" okText="yes" okType="danger">
            <Direction type="primary">危险操作</Direction>
          </Popconfirm>
        </div>
      </Theme>
    </Wrapper>
  );
};
export default () => {
  return [
    <WrapperDemo />,
    <Wrapper>
      <Condition />
    </Wrapper>,
  ];
};
