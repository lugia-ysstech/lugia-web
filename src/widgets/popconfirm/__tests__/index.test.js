/**
 *
 * create by liangguodong
 *
 * @flow
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { WrapperDemo } from '../demo';
import Enzyme, { mount } from 'enzyme';
import 'jest-styled-components';
import Adapter from 'enzyme-adapter-react-16';
import ReactDOM from 'react-dom';
Enzyme.configure({ adapter: new Adapter() });

describe('popconfirm Demo', () => {
  it('Component JSON', () => {
    ReactDOM.createPortal = node => node;
    global.svtest = true;
    const renders = renderer.create(<WrapperDemo />);
    expect(renders.toJSON()).toMatchSnapshot();
    global.svtest = false;
  });
});
