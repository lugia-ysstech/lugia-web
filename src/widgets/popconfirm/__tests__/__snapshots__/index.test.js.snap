// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popconfirm Demo Component JSON 1`] = `
.c2 {
  display: inline-block;
  margin-bottom: 0;
  box-sizing: border-box;
  touch-action: manipulation;
  cursor: pointer;
  white-space: nowrap;
  font-family: 'Trebuchet MS',Arial,Helvetica,sans-serif;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  position: relative;
  text-transform: none;
  outline: 0;
  background-color: #ffffff;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #cccccc;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #cccccc;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #cccccc;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #cccccc;
}

.c4 {
  color: #666666;
  font-size: 1.4rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
}

.c1 {
  display: inline-block;
}

.c6 {
  box-sizing: border-box;
  line-height: 1.5;
  display: inline-block;
  font-family: inherit;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  outline: none;
  height: 100%;
  width: 100%;
  font-size: 1.2rem;
  cursor: text;
  border-top-width: 0.1rem;
  border-top-style: solid;
  border-top-color: #cccccc;
  border-bottom-width: 0.1rem;
  border-bottom-style: solid;
  border-bottom-color: #cccccc;
  border-left-width: 0.1rem;
  border-left-style: solid;
  border-left-color: #cccccc;
  border-right-width: 0.1rem;
  border-right-style: solid;
  border-right-color: #cccccc;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c6::-webkit-input-placeholder {
  color: #cccccc;
  font-size: 1.2rem;
}

.c6::-moz-placeholder {
  color: #cccccc;
  font-size: 1.2rem;
}

.c6:-ms-input-placeholder {
  color: #cccccc;
  font-size: 1.2rem;
}

.c6::placeholder {
  color: #cccccc;
  font-size: 1.2rem;
}

.c5 {
  position: relative;
  display: inline-block;
  outline: none;
  min-width: 3rem;
  width: 100%;
}

.c0 {
  margin: 100px;
}

<div
  className="c0"
>
  <div
    style={
      Object {
        "marginLeft": 80,
        "whiteSpace": "nowrap",
      }
    }
  >
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            TL
          </span>
        </span>
      </button>
    </div>
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            Top
          </span>
        </span>
      </button>
    </div>
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            TR
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    style={
      Object {
        "float": "left",
        "width": 70,
      }
    }
  >
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            LT
          </span>
        </span>
      </button>
    </div>
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            Left
          </span>
        </span>
      </button>
    </div>
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            LB
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    style={
      Object {
        "marginLeft": 320,
        "width": 70,
      }
    }
  >
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            RT
          </span>
        </span>
      </button>
    </div>
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            Right
          </span>
        </span>
      </button>
    </div>
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            RB
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    style={
      Object {
        "clear": "both",
        "marginLeft": 80,
        "whiteSpace": "nowrap",
      }
    }
  >
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            BL
          </span>
        </span>
      </button>
    </div>
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            Bottom
          </span>
        </span>
      </button>
    </div>
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            BR
          </span>
        </span>
      </button>
    </div>
  </div>
  <br />
  <div
    className="c1"
    onBlur={[Function]}
    onFocus={[Function]}
    style={Object {}}
  >
    <span
      className="c5"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "height": "3.2rem",
        }
      }
    >
      <input
        autoFocus={false}
        className="c6"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        placeholder="聚焦弹出"
        size="default"
        style={
          Object {
            "color": "#333333",
            "paddingBottom": "0rem",
            "paddingLeft": "1rem",
            "paddingRight": "3.5rem",
            "paddingTop": "0rem",
          }
        }
        value=""
      />
    </span>
  </div>
  <div
    className="c1"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={Object {}}
  >
    <button
      className="c2"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c3"
      >
        <span
          className="c4"
          style={
            Object {
              "color": "#ffffff",
              "fontSize": "1.4rem",
            }
          }
        >
           悬停
        </span>
      </span>
    </button>
  </div>
  <div
    className="c1"
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
  >
    <button
      className="c2"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      size="default"
      style={
        Object {
          "backgroundColor": "#4d63ff",
          "borderBottomLeftRadius": "0.4rem",
          "borderBottomRightRadius": "0.4rem",
          "borderBottomWidth": 0,
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopLeftRadius": "0.4rem",
          "borderTopRightRadius": "0.4rem",
          "borderTopWidth": 0,
          "height": "3.2rem",
          "paddingBottom": "0rem",
          "paddingLeft": "1.8rem",
          "paddingRight": "1.8rem",
          "paddingTop": "0rem",
          "width": "8rem",
        }
      }
      type="primary"
    >
      <span
        className="c3"
      >
        <span
          className="c4"
          style={
            Object {
              "color": "#ffffff",
              "fontSize": "1.4rem",
            }
          }
        >
          点击
        </span>
      </span>
    </button>
  </div>
  <br />
  <div>
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            提示
          </span>
        </span>
      </button>
    </div>
    <div
      className="c1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <button
        className="c2"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        size="default"
        style={
          Object {
            "backgroundColor": "#4d63ff",
            "borderBottomLeftRadius": "0.4rem",
            "borderBottomRightRadius": "0.4rem",
            "borderBottomWidth": 0,
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopLeftRadius": "0.4rem",
            "borderTopRightRadius": "0.4rem",
            "borderTopWidth": 0,
            "height": "3.2rem",
            "paddingBottom": "0rem",
            "paddingLeft": "1.8rem",
            "paddingRight": "1.8rem",
            "paddingTop": "0rem",
            "width": "8rem",
          }
        }
        type="primary"
      >
        <span
          className="c3"
        >
          <span
            className="c4"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": "1.4rem",
              }
            }
          >
            危险操作
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;
